src/ZoneServer/Skills/Combat/SkillHitInfo.cs

```diff
@@ -99,10 +99,12 @@ public void ApplyKnockBack(ICombatEntity target)
 			if (this.KnockBackInfo == null)
 				throw new InvalidOperationException("Knock back info is not set.");
 
-			this.HitInfo.Type = this.Skill.Data.KnockDownHitType;
+			this.HitInfo.Type = this.KnockBackInfo.HitType;
 			target.Position = this.KnockBackInfo.ToPosition;
 
 			target.AddState(StateType.KnockedBack, this.KnockBackInfo.Time);
+			if (this.HitInfo.Type == HitType.KnockDown)
```

Comment:
If you wanted to add a knock down lock state, presumably to differentiate them, why add both states here?

Reply:
My interpretation is that knockdown is kind of like a more severe version of knockback.  I think it makes sense that if certain skill or something needs to check if you're in knockback state, it should also accept a knockdown, but not vice-versa.

Reply:
I won't argue that your interpretation doesn't make sense, but it does go against the grain a little, because games often don't care about the difference between a KB and a KD. Especially when it comes to locks, which these calls are all about. The reason this bothers me is that your code _suggests_ that there is a difference, which we technically don't know yet afaik, As such, I'd suggest to comment it, to clarify why we're doing what we're doing here.

Reply:
I added a comment.  If you think it doesn't make sense to do it this way you can also change it.  It's true that currently, we're not aware of anything that needs this distinction.